src/GraphMetadataViewExtension/GraphMetadataViewModel.cs

```diff
@@ -128,21 +127,41 @@ public GraphMetadataViewModel(ViewLoadedParams viewLoadedParams, GraphMetadataVi
             InitializeCommands();
         }
 
+        /// <summary>
+        /// This event is triggered when a new workspace is opened or when the current workspace is cleared.
+        /// This event manages state of the workspace properties (ie GraphDescription, GraphAuthor, HelpLink, Thumbnail)
+        /// as well as the custom properties in the extension which do not live in the workspace model.
+        /// </summary>
         private void OnCurrentWorkspaceChanged(Graph.Workspaces.IWorkspaceModel obj)
         {
-            if (!(obj is HomeWorkspaceModel hwm))
+            //Todo review if the workspace properties should be managed in the Workspace model.
+            //Due to the fact that Dynamo often leaves the workspace objects in memory and resets their properties when you open a new workspace,
+            //the management of state is not straightforward. However it may make more sense to update those properties with the clearing logic.
+
+            //Handle the case of a custom workspace model opening
+            if (obj is not HomeWorkspaceModel hwm)
```

Comment:
this seems confusing - if the obj is not a `HomeWorkspaceModel` -why declare it as `hwm`?

Reply:
yeah. I modified this from `!(obj is Home...` to use the new `is not`, but I guess it's debatable whether that actually makes it cleaner _in this particular instance_.